### ISSUE TITLE
Tinker's contruct Crafting station and Recipe Stage error

### ISSUE THREAD

**Issue by TheLynk** [SlimeKnights/TinkersConstruct#4522]
**Issue description:For example I have blocked all the craft of 'minecraft' with a stage and even if we have the stage we cannot craft an object in it and it writes a message in the echat saying to contact the two modders. I have already contacted the recipe stage modder who says he can do nothing unfortunately**

**If crashed, link to crash report (use a site such as pastebin):**

**Versions:**
* Minecraft:1.16.5
* Forge: 36.1.0
* Mantle: 1.16.5-1.6.103
* Tinkers Construct: 1.16.5-3.0.4.171

**Confirm below that this issue is not covered on the roadmap or "Whats New?" : Yes I confirm**

**Can it be reproduced with *just* Tinkers Construct? If not, list the other mods *required* to reproduce the issue.**
list of the mod : 
- Bookshelf-1.16.5-10.1.11.jar
- CraftTweaker-1.16.5-7.1.0.313.jar
- GameStages-1.16.5-7.1.6.jar
- JEITweaker-1.16.5-1.0.1.27.jar
- jei-1.16.5-7.7.0.104.jar
- Mantle-1.16.5-1.6.103.jar
- RecipeStages-2.0.0.16.jar
- TConstruct-1.16.5-3.0.4.171.jar

Should be fewer than 10 mods, you should be running no mods besides those 10 when you reproduce the issue.

Here is a link with the test curse modpack to reproduce the bug with a map already ready: https://mega.nz/file/VJgWyb7D#XpCZ4x3HLoRjuS12c1tmFHCLFsJExlxCMipPCBzEg8g

**Comment by North-West-Wind** [comment]
I'm also having this problem and CraftTweaker gave me this line:
```
[23:10:47.630][DONE][SERVER][ERROR] Cannot craft staged recipes in inventory: `slimeknights.tconstruct.tables.tileentity.table.crafting.CraftingInventoryWrapper@783bf07c` as we have no access to a player or a container! Please report this to RecipeStages and the offending mod!
```

**Comment by KnightMiner** [comment]
Yes, this is known. No, it is not easy to fix, as in my opinion crafting tables should not have player sensitive recipes. I will look into it again when I have more time.

**Comment by TheLynk** [comment]
I found this mod which takes the concept of your craft table and suddenly why not ask how he does it and in addition I tested there is no bug on it and some functionality to improve for the connection to the storage: https://www.curseforge.com/minecraft/mc-mods/crafting-station

**Comment by KnightMiner** [comment]
He does not take the concept of our crafting station, he takes all our old 1.12 code that lead to a ton of duplication bugs on servers. I intentionally rewrote that 1.12 code because to me duplication bugs are a bigger problem than game stages.

I don't need help reproducing this bug, I know this is an issue and I know exactly why, more comments with more information is not needed. If you care about game stages, disable the crafting station for now, but I don't need more confirmation

**Comment by TheLynk** [comment]
No problem I just wanted to help

**Comment by KnightMiner** [comment]
That commit does what I can to fix this on my side. Will require an update to Recipe Stages when Jared has time. I sent him the relevant information

**Comment by TheLynk** [comment]
Ok and thank you again for your work

**Comment by KnightMiner** [comment]
Fixed in recipe stages 2.0.0.20.

Note its stated as "fixing" in quotation marks, which I assume means there are still some edge cases that Jared could not fully fix.
